### PR TITLE
feat(aws-iot-kinesisstreams): implement new construct

### PR DESCRIPTION
**Pull request by surukonda**
*Issue #, if available:*

*Description of changes:*
* Created new aws-iot-kinesisstreams

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 2ce927be8581278243fc5dbfdb4a509779af6fc3
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a6975c0a-5d31-48f6-aa78-4ec621bf20e0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 75a97fdf1ccb357eb88c398f91fcb3037244333e
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5e234a00-d788-4d03-8041-7706584d4177%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 6fcba59c9213871cca40f07efebbbe06c9f56b48
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f0e67634-da76-47f4-87ca-7a7994deb2b1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — CHANGES_REQUESTED** [review]

File: source/patterns/@aws-solutions-constructs/aws-iot-kinesisstreams/lib/index.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      existingStreamObj: props.existingStreamObj,
+      kinesisStreamProps: props.kinesisStreamProps,
+    });
+
+    // Setup the IAM Role for IoT Actions
+    this.iotActionsRole = new iam.Role(this, 'IotActionsRole', {
+      assumedBy: new iam.ServicePrincipal('iot.amazonaws.com'),
+    });
+
+    // Setup the IAM policy for IoT Actions
```
[the remark this answers is not in the document]
Comment (surukonda) [reply source/patterns/@aws-solutions-constructs/aws-iot-kinesisstreams/lib/index.ts:81]: stream.grantWrite() is adding 3 statements (PutRecord, PutRecords & ListShards). In this case I just need PutRecord. Adding 3 statements wouldn't hurt, but I was being specific. Also iotTopicRule is a CfnResource and doesn't implement IGrantable.

File: source/patterns/@aws-solutions-constructs/aws-iot-kinesisstreams/.eslintignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+lib/*.js
+test/*.js
+*.d.ts
+coverage
+test/lambda/index.js
```
[the remark this answers is not in the document]
Comment (surukonda) [reply source/patterns/@aws-solutions-constructs/aws-iot-kinesisstreams/.eslintignore:5]: removed it

File: source/patterns/@aws-solutions-constructs/aws-iot-kinesisstreams/.gitignore [hunk 3]
```diff
@@ -0,0 +1,16 @@
+lib/*.js
+test/*.js
+!test/lambda/*
```
[the remark this answers is not in the document]
Comment (surukonda) [reply source/patterns/@aws-solutions-constructs/aws-iot-kinesisstreams/.gitignore:3]: removed it

File: source/patterns/@aws-solutions-constructs/aws-iot-kinesisstreams/README.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+|iotTopicRuleProps|[`iot.CfnTopicRuleProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-iot.CfnTopicRuleProps.html)|User provided CfnTopicRuleProps to override the defaults|
+|existingStreamObj?|[`kinesis.Stream`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-kinesis.Stream.html)|Existing instance of Kinesis Stream, providing both this and `kinesisStreamProps` will cause an error.|
+|kinesisStreamProps?|[`kinesis.StreamProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-kinesis.StreamProps.html)|Optional user-provided props to override the default props for the Kinesis data stream, providing both this and `existingStreamObj` will cause an error|
+|createCloudWatchAlarms|`boolean`|Whether to create recommended CloudWatch alarms for Kinesis Data Stream. |
+
+## Pattern Properties
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|iotTopicRule|[`iot.CfnTopicRule`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-iot.CfnTopicRule.html)|Returns an instance of iot.CfnTopicRule created by the construct|
```
[the remark this answers is not in the document]
Comment (surukonda) [reply source/patterns/@aws-solutions-constructs/aws-iot-kinesisstreams/README.md:73]: iotTopicRule cannot be optional, I cannot think of way to forward messages to sqs, lambda or kinesisstreams without creating an iotTopicRule

File: source/patterns/@aws-solutions-constructs/aws-iot-kinesisstreams/README.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+* props [`IotToKinesisStreamsProps`](#pattern-construct-props)
+
+## Pattern Construct Props
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|iotTopicRuleProps|[`iot.CfnTopicRuleProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-iot.CfnTopicRuleProps.html)|User provided CfnTopicRuleProps to override the defaults|
+|existingStreamObj?|[`kinesis.Stream`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-kinesis.Stream.html)|Existing instance of Kinesis Stream, providing both this and `kinesisStreamProps` will cause an error.|
+|kinesisStreamProps?|[`kinesis.StreamProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-kinesis.StreamProps.html)|Optional user-provided props to override the default props for the Kinesis data stream, providing both this and `existingStreamObj` will cause an error|
+|createCloudWatchAlarms|`boolean`|Whether to create recommended CloudWatch alarms for Kinesis Data Stream. |
```
[the remark this answers is not in the document]
Comment (surukonda) [reply source/patterns/@aws-solutions-constructs/aws-iot-kinesisstreams/README.md:67]: I don't know the history, but if the decision is to keep cloudwatchAlarms, I can update the other constructs (may be by next week). Let me know..

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 1722074940b12fe523a77d972713227300fbc1df
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6a5fe62f-779c-4d86-bc52-1e9b6ca8ce8f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — APPROVED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: ba462b52d43bc0d3cd3d89ad14eabd24c2fcf470
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6a85a46b-e44c-4653-bb0a-fda5bf4ef19f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 17391c34927adba2bb837358335dc9a9d802c6b2
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=fbace46a-a26a-4c84-81a2-1052aca2b351%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 9f4499127039eb045143e731730956bf4ed217ce
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ccb927d6-dd84-4ea8-9c34-e1749cd4d13a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — APPROVED** [review]

**Pull request merged**